### PR TITLE
Add GitLab support

### LINKED ISSUE #9: GitLab Support
I currently self-host GitLab and use it almost exclusively, so I'd love to see support for GitLab. GitLab has its [own implementation of webhooks](https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/web_hooks/web_hooks.md) but it I believe it has some compatibility already with GitHub webhooks (you might be able to use a similar request body).

I'll have a look at getting it running with GitLab and make a PR if any changes need to be made.

### PR DESCRIPTION
**Pull request by hugomd**
Closes #9.

~~I'm using a rather _long_ XOR statement in `src/core.js` to check that either `GITHUB_TOKEN` _or_ `GITLAB_TOKEN` are set, the same with `GITHUB_WEBHOOK_SECRET` and `GITLAB_WEBHOOK_SECRET` — not sure if there's a nicer way to do this?~~ I realised that it makes more sense to allow one or both, which makes it easy to clean up the if statement 👍 

Any all and all feedback welcome ✌️

### CODE REVIEW

**Comment by hugomd** [comment]
I can confirm that this works for both GitLab and GitHub without issue 🎉

**Comment by zpnk** [comment]
Hey @hugomd this is awesome, thanks so much for this feature! 🙌

I like that you added nodemon for the dev server. 👍 

Everything looks really good. I want to test this out a little before merging, but from looking at it I don't think there's much I'd change.

File: src/core.js [hunk 1]
```diff
@@ -9,21 +9,19 @@ const axios = require('axios');
 const log = require('./logger');
 const envs = require('./envs');
 
-if (!process.env.GITHUB_TOKEN) throw new Error('GITHUB_TOKEN must be defined in environment. Create one at https://github.com/settings/tokens');
-if (!process.env.GITHUB_WEBHOOK_SECRET) throw new Error('GITHUB_WEBHOOK_SECRET must be defined in environment. Create one at https://github.com/{OWNERNAME}/{REPONAME}/settings/hooks (swap in the path to your repo)');
-if (!process.env.ZEIT_API_TOKEN) throw new Error('ZEIT_API_TOKEN must be defined in environment. Create one at https://zeit.co/account/tokens');
+const env = process.env;
```
Comment (zpnk) [inline src/core.js]: We could probably take this one step further and just destructure the envars out right here:

```js
const {
  GITHUB_TOKEN, 
  GITHUB_WEBHOOK_SECRET, 
  GITLAB_TOKEN,
  GITLAB_WEBHOOK_SECRET,
  ZEIT_API_TOKEN
} = process.env;
```

what do you think?

File: src/core.js [hunk 2]
```diff
@@ -111,6 +114,53 @@ function github({headers, body}) {
...
+      'PRIVATE-TOKEN': env.GITLAB_TOKEN
+    }
+  });
+
+  return {
+    ref: source_branch,
+    sha: id,
+    success: true,
+    name: target.path_with_namespace,
+    alias: `https://${source.name.replace(/[^A-Z0-9]/ig, '-')}-${source_branch.replace(INVALID_URI_CHARACTERS, '-')}.now.sh`,
```
Comment (zpnk) [inline src/core.js]: I think it may be worth it to pull this alias formatting logic out and share it between github/lab.

File: src/core.js [hunk 3]
```diff
@@ -111,6 +114,53 @@ function github({headers, body}) {
...
+    cloneUrl: url.format(Object.assign(
+      url.parse(source.http_url),
+      {auth: `gitlab-ci-token:${env.GITLAB_TOKEN}`}
+    )),
```
Comment (zpnk) [inline src/core.js]: Same for cloneUrl, if the logic can be shared

File: src/server.js [hunk 4]
```diff
@@ -20,7 +20,9 @@ server.post('/', (request, response) => {
   let result;
   try {
     const {headers, body} = request;
-    result = github({headers, body});
+    const keys = Object.keys(headers);
+    if (keys.includes('x-github-event')) result = github({headers, body});
+    if (keys.includes('x-gitlab-event')) result = gitlab({headers, body});
```
Comment (zpnk) [inline src/server.js:25]: Well this was easier than I thought, hah. Nice 👍

**Review by zpnk — COMMENTED** [review]
Just some initial thoughts

File: src/core.js (hunk 1 above)
Comment (hugomd) [reply src/core.js]: That should be much cleaner! 👍

File: src/core.js (hunk 2 above)
Comment (hugomd) [reply src/core.js]: I'll abstract this out ✌️

**Comment by hugomd** [comment]
@zpnk I've made all of the aforementioned changes — abstracted out `cloneUrl` and `alias` formatting, and I changed the  way the env. variables are defined. 

I double checked to see if it works, and it still does! 🎉

**Comment by zpnk** [comment]
@hugomd Thanks for the changes! I'm going to merge this - really appreciate your hard work! 👏

**Pull request merged**